frontend/catalyst/device/qjit_device.py

```diff
@@ -179,6 +179,10 @@ def extract_backend_info(
     if not pathlib.Path(device_lpath).is_file():
         raise CompileError(f"Device at {device_lpath} cannot be found!")
 
+    if hasattr(device, "_track_resources"):
+        # pylint: disable=protected-access
+        device_kwargs["track_resources"] = device._track_resources
+
```

Comment:
This is awkward, we shouldn't have to hardcode knowledge of all device options for each device into Catalyst. Below we make an exception for `braket` because we have no control over it, but that's only a temporary crutch that we shouldn't expand.

Instead, we should make of the proper mechanism (`device_kwargs`) in the Catalyst device interface to pass down constructor arguments: https://docs.pennylane.ai/projects/catalyst/en/stable/dev/custom_devices.html#integration-with-python-devices

Reply:
PR merged into PL. Removed from Catalyst in [fcf05bb](https://github.com/PennyLaneAI/catalyst/pull/1619/commits/fcf05bb3ba790e12df45b01ab0d14a8b01158bc8).

Reply:
Yes, but it'll be for the nightly version that's built tonight, so I will do this tomorrow.